npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb

```diff
@@ -176,13 +180,19 @@ def guess_yarn_version
       def pnpm_version
         return @pnpm_version if defined?(@pnpm_version)
 
-        @pnpm_version = package_manager.locked_version("pnpm") || guess_pnpm_version
+        version = package_manager.requested_version("pnpm") || guess_pnpm_version
+
+        if version && Version.new(version.to_s) < Version.new("7")
+          raise ToolVersionNotSupported.new("PNPM", version.to_s, "7.x, 8.x")
```

Comment:
should this be `7.*` to stick to our pattern?

Reply:
Does it work in `package.json`? If it works, that's fine with me, I just wanted to use something I know it's supported in this particular ecosystem.

Reply:
sorry I meant the error message :) ; I think package.json allows semver.

Reply:
Yes, I meant the error message two!

I'd like to include ranges that make sense for each ecosystem in the error messages, so that users can put that right in their manifests. So, for example in Python, if they want to upgrade to 3.11, they can drop `3.11.*` in their manifest and it will work, but `3.11.x` won't. By using `7.x` here I wanted to follow that approach since I'm not sure if `7.*` is a valid JS version range.

Reply:
Ahh; I think they follow SemVer too so I *assume* (feel free to correct me lol) 7.*.* should work.

Reply:
Tried it and you're right, `7.*` works just the same. I'll change this, thanks for the suggestion!